src/main/java/ch/njol/skript/SkriptCommandTabCompleter.java

```diff
@@ -0,0 +1,102 @@
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.command.TabCompleter;
+import ch.njol.util.StringUtils;
+
+/**
+ * @author Peter Güttinger
+ */
+public class SkriptCommandTabCompleter implements TabCompleter {
+		
+	@Override
+	public List<String> onTabComplete(CommandSender sender, Command command, String alias, String[] args) {
+		ArrayList<String> options = new ArrayList<>();
+		
+		if (!command.getName().equalsIgnoreCase("skript")) {
+			return null;
+		}
+		
+		if (args[0].length() == 0) {
+			options.add("help");
+			options.add("reload");
+			options.add("enable");
+			options.add("disable");
+			options.add("update");
+			options.add("info");
+		}
+		
+		else if (args[0].equalsIgnoreCase("update")) {
+			options.add("check");
+			options.add("changes");
+			options.add("download");
+		}
+		
+		else if (args[0].matches("(?i)(reload|disable|enable)")) {
+			File scripts = new File(Skript.getInstance().getDataFolder(), Skript.SCRIPTSFOLDER);
+			String scriptArg = StringUtils.join(args, " ", 1, args.length); 
+			String fs = File.separator;
+			
+			try {
+				Files.walk(scripts.toPath())
+					.map(Path::toFile)
+					.filter(f -> (!f.isDirectory() && f.getName().toLowerCase().endsWith(".sk")) || f.isDirectory()) // filter folders and skript files only 
+					.filter(f -> { // Filtration for enable, disable and reload
+						if (args[0].equalsIgnoreCase("enable"))
+							return f.getName().startsWith("-");
+						else // reload & disable both accepts only non-hyphened files and not hidden folders
+							return !f.getAbsolutePath().matches(".*?(\\\\-|/-|^-).*") && (f.isDirectory() && !f.getAbsolutePath().matches(".*?(\\\\\\.|/\\.|^\\.).*") || !f.isDirectory());
+					})
+					.filter(f -> { // Autocomplete incomplete script name arg
+						return scriptArg.length() > 0 ? f.getName().startsWith(scriptArg) : true;
+					})
+					.forEach(f -> {
+						if (!f.toString().equals(scripts.toString()))
+							options.add(f.toString()
+								.replace(scripts.toPath().toString() + (!f.isDirectory() && f.getParentFile().toPath().toString().equals(scripts.toPath().toString()) ? fs : ""), "") // Extract file short path, and remove '/' from the beginning of files in root only
+								+ (f.isDirectory() && f.toString().length() > 0 ? fs : "")); // add File.separator at the end of directories
+					}); 
+				
+			// TODO handle file permissions
+			} catch ( IOException e) {
+				e.printStackTrace();
+			}
```

Comment:
This can probably use `ScriptLoader#getLoadedFiles` and `ScriptLoader#getDisabledFiles`

see https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/expressions/ExprScripts.java